Tasks/NuGetInstaller/NuGetInstaller.ps1

```diff
@@ -62,15 +65,40 @@ if(!$nuGetPath)
     $nuGetPath = Get-ToolPath -Name 'NuGet.exe';
 }
 
+if (-not $nugetPath)
+{
+    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+}
+
 if($nuGetRestoreArgs)
 {
+    if($nuGetRestoreArgs.ToLowerInvariant().Contains("-configfile"))
+    {
+        Write-Warning "ConfigFile was passed as a command line parameter, which may be ignored in certain parts of this task. Please specify the config file in the build definition instead."
+    }
+
     $args = ($args + " " + $nuGetRestoreArgs);
 }
 
-
-if (-not $nugetPath)
+if($nugetConfigPath -ne $env:Build_SourcesDirectory)
 {
-    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+    $args = $args + " -configfile " + $tempNuGetConfigPath
+
+    $endpoint = Get-ServiceEndpoint -Context $distributedTaskContext -Name SystemVssConnection
+    if($endpoint.Authorization.Scheme -eq 'OAuth')
+    {
+        Write-Verbose "Getting credentials for $($endpoint)"
```

Comment:
The $() looks unnecessary. Every time I read the line i pause for a couple seconds trying to figure out what I'm overlooking that the $() provides here. Every time I conclude it's necessary. :)